### ISSUE TITLE
January Hackathon Milestones and Deliverables - HuSH+ Data

### ISSUE THREAD

**Issue by KCB13** [ResearchSoftwareInstitute/greendatatranslator#103]
Milestones:

- [x] M0. Draft Fv1 specs with “prototype” data (raw LOINC units, MDCTN coding) & deploy to UCSD, Broad, Jax for comment [Done: @krobasky @xu-hao and Ashok polled the community, not ready to comment on general ML-ready data formats]
- [x] M0.1 “widen” the current data representation [Done: @xu-hao  created a sparse, compressed asci format]
- [x] M0.2 resolve exposome data integration issues (is this PHI?) [Done: @krobasky unable to get movement from NCTraCS, maybe @karafecho can unblock]
- [x] M1. Normalize, clean and map medication codes to gene targets [Student: In process]
- [ ] M2. Harmonize the LOINC units [JAX, unlikely to be completed before hackathon]
- [ ] M3. Deploy D1. and model prototypes to demonstrate utility [@krobasky @xu-hao: In progress]
- [x] M3.1 incorporate comments from M0 & deliver Tv1 + SQL code to github [Done: @xu-hao  checked in code, @krobasky tested]
- [x] M3.2 map asthma severity biomarkers to clinical data fields in D1 [Done: @karafecho provided the relevant biomarker codes: 
- [ ] M3.3 train rpart (supervised) and RUF (unsupervised) machine learning models to D1 and publish results [@krobasky , dependent on M1, which is in progress]
- [x] M4. Create D2, possibly invite externally-published models for inspection and execution against these data [Done: no longer needed based on community feedback]
- [ ] M5. Prototype rnn model against D1; train and tune against D2, validate and report performance [@xu-hao, planned for at the hackathon]
- [ ] M6. Create table xTv1a, an extract of Tv1, details in Kara's comments
- [ ] M7. Create table xTv1a.1, this is the same as xTv1a (M6) with the addition of gene targets

Deliverables: 
- [x] D1. A “wide-format” table of HuSH+ data that can be easily computed over via contemporary ML packages, formatted to support multiple models (e.g., RENCI, UCSD, Broad, and Jax) – we call the format “Fv1” and the table “Tv1”
- [x] D1.1 Curated scripts (postgres SQL on github) for transforming the raw HuSH+ data (json format) into the Fv1 format
- [x] D1.2 A “clean” version of Tv1, still in Fv1 format, with harmonized LOINC units, normalized medication coding, and medication-to-target-gene-mappings; we call this table “Tv2”
- [x] D2. An Fv1-formated table of CLINICAL data against which externally-published models can be run after having been prototyped using HuSH+ data (this table must be fire-walled)
 Should we add “analysis” as a deliverable? Would that be a paper report, email, slides,…?

**Comment by xu-hao** [comment]
Just a clarification M5 (rnn) is not planned for the hackathon. I will likely be working on pitch 15 tools for subsetting the wide table

**Comment by karafecho** [comment]
[ ] M6. Create table Tv1a, a simplified version of the full Tv1. From Tv1, select encounter dates between 01.01.2015 and 12.31.17 and the following column headers:

- Patient ID

- Demographics (HuSH+ birth date, sex, race)

- Encounter date and location (inpatient, outpatient, ED)

-- Vital signs at encounter (BMI, PULSE, RESP, TEMP)

-- ICD codes at encounter, filtered for diagnoses of respiratory conditions and obesity
ICD9 | 493.% | asthma
ICD10 | J45.% | asthma
ICD9 | 464.% | croup
ICD10 | J05.% | croup
ICD9 | 496.% | reactive airway
ICD10 | J44.% | reactive airway
ICD10 | J66.% | reactive airway
ICD9 | 786.% | cough
ICD10 | R05.% | cough
ICD9 | 481.% | pneumonia
ICD9 | 482.% | pneumonia
ICD9 | 483.% | pneumonia
ICD9 | 484.% | pneumonia
ICD9 | 485.% | pneumonia
ICD9 | 486.% | pneumonia
ICD10 | J12.% | pneumonia
ICD10 | J13.% | pneumonia
ICD10 | J14.% | pneumonia
ICD10 | J15.% | pneumonia
ICD10 | J16.% | pneumonia
ICD10 | J17.% | pneumonia
ICD10 | J18.% | pneumonia
ICD9 | 278.00 | obesity
ICD10 | E66.% (except E66.3) | obesity

-- LOINC codes at encounter, filtered for measurements of plasma/serum IgE, circulating eosinophils, circulating neutrophils, and plasma/serum CRP
33536-4 | Miscellaneous allergen IgE Ab RAST class [Presence] in Serum
13834-7 | Total IgE [Units/volume] in Serum by Radioallergosorbent test (RAST)
26449-9 | Eosinophils [#/volume] in Blood
711-2 | Eosinophils [#/volume] in Blood by Automated count
712-0 | Eosinophils [#/volume] in Blood by Manual count
26450-7 | Eosinophils/100 leukocytes in Blood
713-8 | Eosinophils/100 leukocytes in Blood by Automated count
714-6 | Eosinophils/100 leukocytes in Blood by Manual count
26499-4 | Neutrophils [#/volume] in Blood
751-8 | Neutrophils [#/volume] in Blood by Automated count
753-4 | Neutrophils [#/volume] in Blood by Manual count
26511-6 | Neutrophils/100 leukocytes in Blood
770-8 | Neutrophils/100 leukocytes in Blood by Automated count
23761-0 | Neutrophils/100 leukocytes in Blood by   Manual count
1988-5 | C reactive protein [Mass/volume] in Serum or Plasma
30522-7 | C reactive protein [Mass/volume] in Serum or Plasma by High sensitivity   method
11039-5 | C reactive protein [Presence] in Serum or Plasma
35648-5 | C reactive protein [Quintile] in Serum or Plasma by High sensitivity   method
76485-2 | C reactive protein [Moles/volume] in Serum or Plasma
76486-0 | C reactive protein [Moles/volume] in Serum or Plasma by High sensitivity   method
14634-0 | C reactive protein [Titer] in Serum or Plasma
71426-1 | C reactive protein [Mass/volume] in Blood by High sensitivity method

[ ] D3. A simplified table TV1a for use by Green Team during hackathon to begin to explore (state)trait stratification of patients with asthma-like conditions. (See Dropbox/DataTranslatorProject/Slides/TranslatorOverview_v4 12.20.17.)

**Comment by krobasky** [comment]
Very helpful, thanks Kara!
The details in M6 were intended to fall under M3; and I had mistakenly marked this as "done" based on a previous email; I'm very glad to have the additional detail added here.

Some remaining items:

1. LOINC CODES:

a. What to do about severely imbalanced in record-counts?
Here are the tallies I found for the above LOINC codes in the Hush+v1 data. We may find more in the v2, but I would suggest a strategy for dealing with imbalances. For example, many LOINCs are missing (indicating a cel-size of <11 records), I would suggest omitting all LOINCs in the bottom quartile, but we can try as-is and see what happens, if that's preferred.

Tally   Code    Description
832     13834-7 Total IgE

13461   1988-5  C reactive protein [Mass/volume] in Serum or Plasma     C reactive protein
729     30522-7 C reactive protein [Mass/volume] in Serum or Plasma by High sensitivity method

52228   26449-9 Eosinophils[#/volume] in blood
33729   26450-7 Eosinophils/100 leukocytes in blood
102622  711-2   Eosinophils [#/volume] in Blood by Automated count

73301   26499-4 Neutrophils [#/volume]
33840   26511-6 Neutrophils/100 leukocytes in Blood

20      714-6   Eosinophils/100 leukocytes in Blood by Manual count
121398  751-8   Eosinophils/100 leukocytes in Blood by Automated count

b. How do we harmonize units?

We need a strategy for harmonizing units. Among the LOINCS that appear in the HUSH+v1 records, the units follow:

tally     code                   min,max,mean,3*std             flags              units
832     LOINC:13834-7   2.00,5000.00,219.34,1611.18     [H],[L],[A],    kU/L,IU/mL,

13461   LOINC:1988-5    0.00,450.00,10.05,91.49 [H],H,  MG/DL,mg/L,mg/dL,MG/L,
729     LOINC:30522-7   0.10,243.30,5.77,30.38          MG/L,

52228   LOINC:26449-9   0.00,15.90,0.20,1.04    [H],H,  x10 9th/L,10*9/L,/µL,X10E3/uL,
33729   LOINC:26450-7   0.00,9999999.00,299.01,163347.67        [H],[L],[A],    %,L,
102622  LOINC:711-2     0.00,343.00,0.18,4.70   [H],[A],        10*9/L,10^3u/L,x10E3/uL,K/uL,/µL,x10(3)/mcL,

73301   LOINC:26499-4   0.00,122.30,5.29,12.91  H,CL,L, x10 9th/L,
33840   LOINC:26511-6   0.00,9999999.00,359.98,163078.55        [H],[L],[A],    %,H,L,

20      LOINC:714-6     0.00,18.00,3.40,12.99   [H],    %,
121398  LOINC:751-8     0.00,6370.00,5.27,77.14 [L],[H],[A],L,H,CL,     k/uL,10*9/L,10^3u/L,x10E3/uL,K/uL,/µL,x10 9th/L,x10(3)/mcL,H,L,X10E3/uL,

2. Vital signs
How to get these from the HUSH+ data?

**Comment by karafecho** [comment]
New item:

Why are LOINC codes deleted for counts of <11 records?

**Comment by krobasky** [comment]
Following are the medication mappings (also useful for one of the pitches):

Medication names (also listed in this thread above):
/projects/datatrans/krobasky/hackathonJan2018/mednames
RxNorm ids for medication names:
/projects/datatrans/krobasky/hackathonJan2018/map.medname-rxnorm.tsv
Gene names for the medication names:
/projects/datatrans/krobasky/hackathonJan2018/map.medname-genename.tsv
MDCTN codes (from hushplus) for the medication names:
/projects/datatrans/krobasky/hackathonJan2018/map.medname-MDCTN.tsv

**Comment by stevencox** [comment]
Hakathon over.
